src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py

```diff
@@ -1804,6 +1817,348 @@ def test_fuzz_with_mutations_using_ml_rnn(self, mock_execute, *_):
           '/fake/inputs-disk/temp-1337/mutations',
       ]])
 
+  @mock.patch('bot.fuzzers.libFuzzer.launcher.add_recommended_dictionary',
+              lambda x, y, z: False)
+  @mock.patch('multiprocessing.cpu_count', return_value=1)
+  @mock.patch('sys.stdout', new_callable=test_utils.MockStdout)
+  def test_fuzz_with_dataflow_tracing(self, mock_stdout, *_):
+    """Tests fuzzing with dataflow tracing."""
+    #//self.mock.cpu_count.return_value = 1
+    self.mock.do_dataflow_tracing.return_value = True
+
+    self.fs.CreateDirectory('/fake/corpus_dir')
+    self.fs.CreateFile('/fake/testcase_basic')
+    self.fs.CreateDirectory('/fake/dfsan_build')
+    self.fs.CreateFile('/fake/dfsan_build/fake_fuzzer')
+
+    os.environ['FUZZ_CORPUS_DIR'] = '/fake/corpus_dir'
+    os.environ['DATAFLOW_BUILD_DIR'] = '/fake/dfsan_build'
+
+    new_units_added = 11
+    with mock.patch(
+        'subprocess.Popen',
+        create_mock_popen(self.no_crash_output,
+                          '/fake/inputs-disk/temp-1337/new', '/fake/corpus_dir',
+                          new_units_added)) as mock_popen:
+      launcher.main([
+          'launcher.py',
+          '/fake/testcase_basic',
+          'fake_fuzzer',
+          '-max_len=80',
+      ])
+
+      self.assertEqual(mock_popen.commands, [[
+          '/fake/build_dir/fake_fuzzer', '-max_len=80', '-rss_limit_mb=2048',
+          '-timeout=25', '-collect_data_flow=/fake/dfsan_build/fake_fuzzer',
```

Comment:
Can we check that `'-collect_data_flow=/fake/dfsan_build/fake_fuzzer'` is in the commands rather than compare against all commands?

Reply:
Sure, but better refactor all the tests at once. Or maybe even re-write them if they won't really work after the fuzzing engine pluggability work.